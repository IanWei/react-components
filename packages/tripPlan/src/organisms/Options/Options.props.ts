export interface OptionsProps {
    label: string,
    options: string[],
    handleCheckboxChange(label: string): void
}
