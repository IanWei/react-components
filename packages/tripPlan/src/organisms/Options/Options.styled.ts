import styled from 'styled-components';
import { spacing } from '../../foundation/spacing';
import { Description } from '../../atoms';

export const StyledOptionWrapper = styled.div`
    display: flex;
    flex-direction: column;
`
export const StyledOptionLabel = styled(Description)`
    margin-bottom: ${spacing.xxxs};
`
