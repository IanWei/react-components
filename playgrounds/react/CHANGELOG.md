# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.2.0](https://github.com/IanWei/react-components/compare/v0.1.1...v0.2.0) (2021-07-04)


### Features

* calendar component ([24182ce](https://github.com/IanWei/react-components/commit/24182ce40d531de8a861fa57808c2b16e24e54f9)), closes [#031](https://github.com/IanWei/react-components/issues/031)
* calendar continue ([38cd923](https://github.com/IanWei/react-components/commit/38cd9238ce43aef2df0acaf1fb614f53809359bf)), closes [#033](https://github.com/IanWei/react-components/issues/033)
* keyboard acceisible ([929915f](https://github.com/IanWei/react-components/commit/929915fd32803440d14a673b550c575a0bc37828)), closes [#028](https://github.com/IanWei/react-components/issues/028)
* navigate component ([fd3c74a](https://github.com/IanWei/react-components/commit/fd3c74aa7ed9e52c12b1b904a7658caa8ec1f119)), closes [#032](https://github.com/IanWei/react-components/issues/032)
* update Select component ([356c718](https://github.com/IanWei/react-components/commit/356c718b96c5675da9d345a9b8aee2c369a3af9a)), closes [#027](https://github.com/IanWei/react-components/issues/027)
