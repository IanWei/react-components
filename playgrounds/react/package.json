{
  "name": "@playground/react",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "dev": "parcel src/index.html -p 3000"
  },
  "devDependencies": {
    "@types/react": "^17.0.13",
    "@types/react-dom": "^17.0.8",
    "parcel-bundler": "^1.12.5",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "typescript": "^4.3.5"
  },
  "private": true
}
