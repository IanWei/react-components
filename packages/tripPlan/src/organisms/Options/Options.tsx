import React, { FC } from 'react';
import { OptionsProps } from './Options.props';
import { StyledOptionLabel, StyledOptionWrapper } from './Options.styled';
import { Checkbox } from '../../molecules/Checkbox';

export const Options: FC<OptionsProps> = ({ label, options, handleCheckboxChange}) => {
    return (
        <StyledOptionWrapper>
            <StyledOptionLabel>{label}</StyledOptionLabel>
            {options.map((option) => (
                <Checkbox key={option} label={option} handleCheckboxChange={handleCheckboxChange}/>
            ))}
        </StyledOptionWrapper>
    )
}
