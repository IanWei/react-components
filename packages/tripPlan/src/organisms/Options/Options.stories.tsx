import React from 'react';
import { Options } from './Options';
import { OptionsProps } from './Options.props';
import { Story } from '@storybook/react';

export default {
    title: 'Organisams/Options',
    component: Options
}

const Template: Story<OptionsProps> = (args) => <Options {...args}/>

export const Base: Story<OptionsProps> = Template.bind({});

Base.args = {
    label: 'Mode',
    options: ['Metro', 'Train', 'Bus', 'Ferry', 'Light Rail']
}
