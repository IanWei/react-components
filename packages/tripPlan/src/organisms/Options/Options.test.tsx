import React from 'react';
import { render, screen } from '@testing-library/react';
import { Options } from './Options';

describe('Options', () => {
    let optionsElement: HTMLDivElement;
    let allOptions: HTMLInputElement[];
    let snapshot: () => DocumentFragment;
    const label = 'priority';
    const options = ['High', 'Normal', 'Low'];
    const handleCheckboxChange = jest.fn();

    beforeEach(() => {
        const { queryByTitle, queryAllByRole, asFragment } = render(
            <Options
                label={label}
                options={options}
                handleCheckboxChange={handleCheckboxChange}
            />
        );
        optionsElement = queryByTitle('options') as HTMLDivElement;
        allOptions = queryAllByRole('checkbox') as HTMLInputElement[];
        snapshot = asFragment;
    });

    it('should render options component', () => {
        expect(optionsElement).toBeTruthy();
    });

    it('should render all options', () => {
        expect(allOptions).toHaveLength(3);
    });

    it('should render label passed by props', () => {
        expect(screen.queryByTitle('option-title')).toHaveTextContent(
            'priority'
        );
    });

    it('should test the snapshot of the component', () => {
        expect(snapshot()).toMatchSnapshot();
    });
});
