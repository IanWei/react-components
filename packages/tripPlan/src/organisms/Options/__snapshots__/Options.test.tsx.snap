// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options should test the snapshot of the component 1`] = `
<DocumentFragment>
  <div
    class="sc-hKFxyN jfpiyA"
    title="options"
  >
    <p
      class="sc-dlnjwi sc-eCApnc ea-dfCT cqGmaj"
      role="option-label"
    >
      priority
    </p>
    <div
      class="sc-jSFjdj dETBGi"
    >
      <label
        class="sc-gKAaRy dOQByB"
        role="label"
      >
        <input
          aria-checked="false"
          aria-label="High"
          class="sc-iCoGMd cHMfVY"
          role="checkbox"
          type="checkbox"
        />
        High
      </label>
    </div>
    <div
      class="sc-jSFjdj dETBGi"
    >
      <label
        class="sc-gKAaRy dOQByB"
        role="label"
      >
        <input
          aria-checked="false"
          aria-label="Normal"
          class="sc-iCoGMd cHMfVY"
          role="checkbox"
          type="checkbox"
        />
        Normal
      </label>
    </div>
    <div
      class="sc-jSFjdj dETBGi"
    >
      <label
        class="sc-gKAaRy dOQByB"
        role="label"
      >
        <input
          aria-checked="false"
          aria-label="Low"
          class="sc-iCoGMd cHMfVY"
          role="checkbox"
          type="checkbox"
        />
        Low
      </label>
    </div>
  </div>
</DocumentFragment>
`;
